### PR TITLE
[ci] M1 release

### PR DESCRIPTION
**Pull request by Leonz5288**
Related issue = #2254

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -81,6 +85,53 @@ jobs:
...
+      run:
+        shell: /bin/bash -l {0}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+
+      - name: Build
+        run: |
+          conda activate $PYTHON
```
Comment (ailzhang) [inline .github/workflows/release.yml:104]: if conda activate worked, the python executable should have been callable without adding to PATH?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -81,6 +85,53 @@ jobs:
...
+          export PATH=/Users/github/miniforge3/envs/$PYTHON/bin:$PATH
+          arch -arm64 python3 -m pip uninstall taichi -y
```
Comment (ailzhang) [inline .github/workflows/release.yml:106]: No need to set arch -arm64 anymore, see #2845

**Review by ailzhang — COMMENTED** [review]
Nice job! Thanks! Only a few nits in the comments.

File: .github/workflows/release.yml (hunk 2 above)
Comment (Leonz5288) [reply .github/workflows/release.yml:106]: I tried the default trick, but it does not work for the bash I use...

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -81,6 +85,51 @@ jobs:
...
+      run:
+        shell: "/usr/bin/arch -arch arm64e /bin/bash --noprofile --norc -eo pipefail {0}"
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+
+      - name: Build
+        run: |
+          export PATH=/Users/github/miniforge3/envs/$PYTHON/bin:$PATH
```
Comment (k-ye) [inline .github/workflows/release.yml:104]: Is this how conda is enabled 😅  ?

Comment (Leonz5288) [reply .github/workflows/release.yml:104]: Well... This is complicated... We can use conda activate $PYTHON. However, this won't change the PATH to python for some reason. Even if we activate conda env, we still need to manually set python path... So we can just get rid of the activate.

Comment (Leonz5288) [reply .github/workflows/release.yml:104]: BTW, the conda related github actions all requires setting PATH

**Review by ailzhang — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**